modules/auxiliary/gather/ldap_query.rb

```diff
@@ -206,15 +301,113 @@ def output_data_csv(entries)
     generate_rex_tables(entries, 'csv')
   end
 
-  def normalize_entries(entries)
+  def query_attributes_data(ldap, entry_keys)
+    # def perform_ldap_query(ldap, filter, attributes, base: nil)
+    filter = '(|'
+    entry_keys.each do |key|
+      filter += "(LDAPDisplayName=#{key})"
+    end
+    filter += ')'
+    attributes = ['LDAPDisplayName', 'isSingleValued', 'oMSyntax', 'attributeSyntax']
+    attributes_data = perform_ldap_query(ldap, filter, attributes, base: ['CN=Schema,CN=Configuration', @base_dn].join(','))
+
+    entry_list = []
+    for entry in attributes_data do
+      data = {}
+      data[:ldapdisplayname] = entry[:ldapdisplayname][0]
+      if entry[:issinglevalued][0] == 'TRUE'
+        data[:issinglevalued] = 1
+      else
+        data[:issinglevalued] = 0
+      end
+      data[:omsyntax] = entry[:omsyntax][0].to_i
+      data[:attributesyntax] = entry[:attributesyntax][0]
+      entry_list << data
+    end
+    entry_list
+  end
+
+  def normalize_entries(ldap, entries)
     cleaned_entries = []
     entries.each do |entry|
       # Convert to a hash so we get only the data we need.
       entry = entry.to_h
       entry_keys = entry.keys
-      entry_keys.each do |key|
-        entry[key] = entry[key].map { |v| Rex::Text.to_hex_ascii(v) }
+      entry_keys.sort! # Sort entry keys alphabetically since LDAP results are returned alphabetically.
+      entry_list = query_attributes_data(ldap, entry_keys)
+      entry_list.each do |attribute_entry|
+        attribute_name = attribute_entry[:ldapdisplayname].to_s.downcase.to_sym
+        modified = false
+        case attribute_entry[:omsyntax]
+        when 1 # Boolean
+          entry[attribute_name][0] = entry[attribute_name][0] == 1
```

Comment:
Booleans are almost always true for non zero values.

```suggestion
          entry[attribute_name][0] = entry[attribute_name][0] != 0
```

Reply:
Fixed with https://github.com/rapid7/metasploit-framework/pull/16979/commits/be95e69e722cf8950502f5c5562ba2e76d01c872